src/main/java/com/javadeobfuscator/deobfuscator/rules/special/RuleSuperblaubeereObfuscation.java

```diff
@@ -0,0 +1,218 @@
+                            clinit.instructions.remove(clinit.instructions.getFirst());
+                            classNode.fields.remove(field);
+                            for (MethodNode method : classNode.methods) {
+                                for (AbstractInsnNode ain : method.instructions.toArray()) {
+                                    if (ain.getOpcode() == Opcodes.GETSTATIC && ((FieldInsnNode) ain).name.equals(field.name)
+                                        && ((FieldInsnNode) ain).desc.equals(field.desc)
+                                        && ((FieldInsnNode) ain).owner.equals(classNode.name)) {
+                                        if (Utils.isInteger(ain.getNext())) {
+                                            return "Found potential number pool in class " + classNode.name;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                // String encryption
+                stringEncrypt:
+                {
+                    if (first != null && first.getOpcode() == Opcodes.INVOKESTATIC && ((MethodInsnNode) first).desc.equals("()V")
+                        && ((MethodInsnNode) first).owner.equals(classNode.name)) {
+                        MethodNode refMethod = classNode.methods.stream()
+                                .filter(m -> m.name.equals(((MethodInsnNode) first).name) && m.desc.equals("()V")).findFirst().orElse(null);
+                        if (refMethod == null) {
+                            break stringEncrypt;
+                        }
+                        AbstractInsnNode methodFirst = refMethod.instructions.getFirst();
+                        if (methodFirst.getOpcode() == Opcodes.NEW && ((TypeInsnNode) methodFirst).desc.equals("java/lang/Exception")
+                            && methodFirst.getNext() != null && methodFirst.getNext().getOpcode() == Opcodes.DUP
+                            && methodFirst.getNext().getNext() != null && methodFirst.getNext().getNext().getOpcode() == Opcodes.INVOKESPECIAL
+                            && ((MethodInsnNode) methodFirst.getNext().getNext()).name.equals("<init>")
+                            && ((MethodInsnNode) methodFirst.getNext().getNext()).owner.equals("java/lang/Exception")
+                            && methodFirst.getNext().getNext().getNext() != null
+                            && methodFirst.getNext().getNext().getNext().getOpcode() == Opcodes.INVOKEVIRTUAL
+                            && ((MethodInsnNode) methodFirst.getNext().getNext().getNext()).name.equals("getStackTrace")
+                            && (((MethodInsnNode) methodFirst.getNext().getNext().getNext()).owner.equals("java/lang/Exception")
+                                || ((MethodInsnNode) methodFirst.getNext().getNext().getNext()).owner.equals("java/lang/Throwable"))
```

Comment:
Some of the deobfuscator's instruction utilities might be a better tool for making this less verbose.

Reply:
Very likely, all I did was basically ctrl-c ctrl-v from the transformer so far. Will improve this.

Reply:
Yeah, looks like it.  It would probably be a great idea to that and related transformers cleaned up as well.

Reply:
Code of superblaubeere heavily improved.